Patch-Source: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/c34e382560ab8858e9f9eec5e9a8b56b0dcbde7f.patch
From c34e382560ab8858e9f9eec5e9a8b56b0dcbde7f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 20 Sep 2024 00:33:29 +0200
Subject: [PATCH] status/keyboard: Fix execution order when activating source

When commit ce89b15bb123d made the code async, it did not only
delay releasing the keyboard until after the engine has been
updated, but also the following code that updates the current
input source.

One result is that the current source is now initialized later,
which breaks any code that relies on the source being set.

This affects the login screen in particular (which uses different
`InputSourceSettings` than the regular session): It fails to come
up entirely if the OSK is enabled.

To fix the issue, update the current source before awaiting the
ibus engine update.

Fixes: ce89b15bb1 ("ibusManager: Use async await instead of callbacks")
Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7912
---
 js/ui/status/keyboard.js | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/js/ui/status/keyboard.js b/js/ui/status/keyboard.js
index d900d3ed718..46d06de7db4 100644
--- a/js/ui/status/keyboard.js
+++ b/js/ui/status/keyboard.js
@@ -519,14 +519,14 @@ export class InputSourceManager extends Signals.EventEmitter {
         else
             engine = 'xkb:us::eng';
 
-        await this._ibusManager.setEngine(engine);
-        if (!this._reloading)
-            KeyboardManager.releaseKeyboard();
-
         this._currentInputSourceChanged(is);
 
         if (interactive)
             this._updateMruSettings();
+
+        await this._ibusManager.setEngine(engine);
+        if (!this._reloading)
+            KeyboardManager.releaseKeyboard();
     }
 
     _updateMruSources() {
-- 
GitLab

